addons/l10n_br_website_sale/views/templates.xml

```diff
@@ -33,6 +33,47 @@
                 </div>
             </t>
         </div>
+        <!-- o_city must remain in DOM, otherwise the standard website_sale js breaks. -->
+        <input id="o_city" position="attributes">
+            <attribute name="class" separator=" " add="o_standard_address"/>
+        </input>
+        <input id="o_city" position="after">
+            <select id="o_city_id" name="city_id" class="form-select o_extended_address">
+                <option value="">City...</option>
+                <t t-foreach="cities" t-as="c">
+                    <option t-att-value="c.id" t-att-selected="c.id == city.id" t-att-code="c.id" t-att-state-id="c.state_id.id" t-att-zip-ranges="c.l10n_br_zip_ranges">
```

Comment:
IMO it's bad practice to hardcode cities that way. Who's gonna maintain this list? 

I don't know the probability of a change in the cities in Brazil and I expected it to be low but if a change in the zip range occurs, it will prevent customers from buying on our eCommerce. 

On some eCommerce, an extra step or a model suggests an address (Something like: "Did you mean this address: [...]"), that gives the customer the choice to keep what he entered. Also, did you check if any external API exists to check the cities?

Reply:
Hello @chevalierv thanks for the feedback!
We need the city field as it is a required attribute to be sent to Avalara in the request for tax computation and EDI-invoice.

In terms of updates in cities, based on my research, in the last 20 years, there were 8 major changes (i.e. cities that changed their name drastically), and 31 total changes (including accent additions for instance). And from 2017 there were 7 changes (where 3 were major changes). [Here's the file](https://docs.google.com/spreadsheets/d/16EywGNGGHwjK2o6ZaU4G15_O9112nrsQXctcbeSAXl0/edit?gid=171335073#gid=171335073) for your reference.

In the other big eCommerce websites, they have the city field as a list, and what they do is also the zip mapping (given the range, we can deduce the city, and we do that now). But not sure if your idea is to have an editable city field that could be mapped with a city in the city_id in the backend (?) but maybe overkilling if we use the zip to map anyway.

All changes in cities per year from 2000:
![image](https://github.com/user-attachments/assets/0a0e94ac-389f-4ae4-a23a-8a07123ec743)

Major changes from 2000:
![image](https://github.com/user-attachments/assets/bb21119f-8bbb-48d1-9f94-9f05f7ae3a90)

cc' @jorenvo

Reply:
Hi Valentin, thanks for taking a look! I agree with @vbe-odoo, changes should be minimal. We looked at some APIs but they come with maintenance as well. A random free API may change, get shut down, impose rate limits on customers with lots of eCommerce traffic, ban IPs etc.